codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -46,24 +50,20 @@
  */
 public class DefaultHeaders<K, V, T extends Headers<K, V, T>> implements Headers<K, V, T> {
     /**
-     * How big the underlying array is for the hash data structure.
-     * <p>
-     * This should be a power of 2 so the {@link #index(int)} method can full address the memory.
+     * Enforce an upper bound of 128 because {@link #hashMask} is a byte.
+     * The max possible value of {@link #hashMask} is one less than this value.
      */
-    private static final int ARRAY_SIZE = 1 << 4;
-    private static final int HASH_MASK = ARRAY_SIZE - 1;
-    static final int HASH_CODE_SEED = 0xc2b2ae35; // constant borrowed from murmur3
+    private static final int ENTRIES_MAX_SIZE = min(128,
```

Comment:
Just the underlying array which is used to build the hash data structure. The motivation for this PR is to be able to preserve memory in the event that there are only a small amount of headers.  To support this the underlying array size is dynamically sizable (up to 128), so we only need an additional `byte` to track the mask.

Before this PR the size is fixed at 16. So this only provides more flexibility. Do you have a use cases where you would like the underlying hash array to be larger than 128?